numpy/__init__.py

```diff
@@ -177,19 +177,65 @@
     warnings.filterwarnings("ignore", message="numpy.ufunc size changed")
     warnings.filterwarnings("ignore", message="numpy.ndarray size changed")
 
-    # oldnumeric and numarray were removed in 1.9. In case some packages import
-    # but do not use them, we define them here for backward compatibility.
-    oldnumeric = 'removed'
-    numarray = 'removed'
-
-    # We don't actually use this ourselves anymore, but I'm not 100% sure that
-    # no-one else in the world is using it (though I hope not)
-    from .testing import Tester
-
-    # Pytest testing
-    from numpy._pytesttester import PytestTester
-    test = PytestTester(__name__)
-    del PytestTester
+
+    if sys.version_info[0] >= 3 and sys.version_info[1] >= 7:
+        import importlib
+        _get_attr_warning_message = (
+            "Importing {attr} from numpy is deprecated and will be removed "
+            "in numpy {version_removed}."
+        )
+
+        def __getattr__(attr):
+            if attr == 'testing':
+                # lazy import testing since it is a costly module to import
+                warnings.warn(
+                    _get_attr_warning_message.format(attr=attr,
+                                                     version_removed='1.22') +
+                    " Please use ``import numpy.testing as testing`` instead.",
+                    DeprecationWarning,
+                    stacklevel=2)
+                return importlib.import_module("." + attr, __name__)
```

Comment:
As Eric said, just use import, importlib is just harder to read.

Reply:
Yeah. I guess I just didn't understand the point of the PEP that had the importlib example